### PR TITLE
MOB-1397 - Created Push API service.

### PR DESCRIPTION
**Pull request by Oleg-Pecheneg**
Also setup core architecture for messaging functionality

### CODE REVIEW

File: unstoppable-ios-app/domains-manager-ios/Entities/Wallets/UDWallet.swift [hunk 1]
```diff
@@ -50,6 +50,17 @@ enum BlockchainType: String, CaseIterable, Codable, Hashable {
             return "Polygon"
         }
     }
+    
+    func supportedChainId(isTestNet: Bool) -> Int? {
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Entities/Wallets/UDWallet.swift:54]: This method would belong better to `UnsConfigManager.swift`

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/Cells/ChatTextCell/ChatTextCell.swift [hunk 2]
```diff
@@ -14,7 +14,7 @@ final class ChatTextCell: UICollectionViewCell {
     @IBOutlet private weak var timeLabel: UILabel!
     
     private var textViewSideConstraints: [NSLayoutConstraint] = []
-    private var sender: ChatSender?
+    private var sender: MessagingChatSender?
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/Cells/ChatTextCell/ChatTextCell.swift:17]: Isn't `MessagingChat` redundant? Simply `ChatSender` was better, if there are not specific reason for this

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/ChatsList/Cells/ChatListCell/ChatListCell.swift [hunk 3]
```diff
@@ -51,24 +51,27 @@ extension ChatListCell {
...
     }
     
-    func lastMessageTextFrom(messageType: ChatMessageType) -> String  {
-        switch messageType {
-        case .text(let message):
-            return message.text
-        }
+    func lastMessageTextFrom(message: MessagingChatMessageDisplayInfo?) -> String  {
+        ""
+//        switch messageType {
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/Messaging/ChatsList/Cells/ChatListCell/ChatListCell.swift:72]: Dead?

File: unstoppable-ios-app/domains-manager-ios/Services/MessagingService/MessagingServiceProtocol.swift [hunk 4]
```diff
@@ -9,7 +9,36 @@ import Foundation
...
+                                  page: Int,
+                                  limit: Int) async throws -> [MessagingChatDisplayInfo]
+    func getMessagesForChat(_ chat: MessagingChatDisplayInfo,
+                            fetchLimit: Int) async throws -> [MessagingChatMessageDisplayInfo]
+    func sendMessage(_ messageType: MessagingChatMessageDisplayType,
+                     in chat: MessagingChatDisplayInfo) throws -> MessagingChatMessageDisplayInfo
+    func makeChatRequest(_ chat: MessagingChatDisplayInfo, approved: Bool) async throws
 }
+
+//protocol MessagingServiceListener: AnyObject {
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/MessagingService/MessagingServiceProtocol.swift:25]: Dead?

**Review by rommex — APPROVED** [review]
With minor questions and overall suggestion to trim 'MessagingChat' naming as it sounds like 'buttery butter'

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/Cells/ChatTextCell/ChatTextCell.swift (hunk 2 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Chat/Cells/ChatTextCell/ChatTextCell.swift:17]: I've decided to make naming convention for all messaging-related entities to start with Messaging. MessagingChat, MessagingChatMessage, etc. It felt confusing with old naming convention in my experience .

File: unstoppable-ios-app/domains-manager-ios/Entities/Wallets/UDWallet.swift (hunk 1 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Entities/Wallets/UDWallet.swift]: It's appear this function wasn't used at all. Removed

**Pull request merged**